Source/WebKit/WPEPlatform/wpe/drm/WPEScreenDRM.cpp

```diff
@@ -67,9 +70,14 @@ WPEScreen* wpeScreenDRMCreate(std::unique_ptr<WPE::DRM::Crtc>&& crtc, const WPE:
     auto* priv = WPE_SCREEN_DRM(screen)->priv;
     priv->crtc = WTFMove(crtc);
 
-    double scale = 1;
+    double scale;
     if (const char* scaleString = getenv("WPE_DRM_SCALE"))
         scale = g_ascii_strtod(scaleString, nullptr);
+    else {
+        if (!priv->wpeSettings)
+            priv->wpeSettings = adoptGRef(WPE_SETTINGS(g_object_new(WPE_TYPE_SETTINGS, nullptr)));
```

Comment:
We shouldn't create the setting here either. Again, we should get the settings from the display. Since we don't have access to the display, we could move the scale initialization to the WPEDisplayDRM, right after the screen is created. Maybe we could also move this part to a different patch.